cookbooks/fb_systemd/recipes/networkd.rb

```diff
@@ -24,6 +24,10 @@
   service svc do
     only_if { node['fb_systemd']['networkd']['enable'] }
     subscribes :restart, 'package[systemd packages]', :immediately
+    if svc == 'systemd-networkd.socket'
+      notifies :stop, 'service[systemd-networkd.service]', :before
+      notifies :start, 'service[systemd-networkd.service]', :immediately
+    end
```

Comment:
Ah, thanks! We hit this internally as well on Ubuntu, and I had an in-progress diff to skip the socket here, but your approach is much better.